src/main/java/org/cascadebot/cascadebot/commands/fun/CatCommand.java

```diff
@@ -25,37 +25,37 @@
-                if(member.getUser().getIdLong() != catButtons.getOwner().getUser().getIdLong()) {
-                    return;
-                }
-                try {
-                    if (message.getEmbeds().size() > 0) {
-                        EmbedBuilder embedBuilder = MessagingObjects.getClearThreadLocalEmbedBuilder();
-                        embedBuilder.setImage(getCatUrl());
-                        message.editMessage(embedBuilder.build()).queue();
-                    } else {
-                        context.getUIMessaging().replyImage(getCatUrl()).thenAccept(catMessage -> {
-                            catButtons.addButtonsToMessage(catMessage);
-                            catButtons.setMessage(catMessage.getIdLong());
-                            context.getData().addButtonGroup(context.getChannel(), catMessage, catButtons);
-                        });
-                        message.delete().queue();
-                    }
-                } catch (IOException e) {
-                    message.editMessage("Error loading cat picture " + UnicodeConstants.FROWNING).queue();
-                }
-            }));
+        ButtonGroup catButtons = new ButtonGroup(context.getUser().getIdLong(), context.getChannel().getIdLong(), context.getGuild().getIdLong());
+        catButtons.addButton(new Button.UnicodeButton(UnicodeConstants.REPEAT, (member, channel, message) -> {
+            if (member.getUser().getIdLong() != catButtons.getOwner().getUser().getIdLong()) {
+                return;
+            }
             try {
-                context.getUIMessaging().replyImage(getCatUrl()).thenAccept(message -> {
-                    catButtons.addButtonsToMessage(message);
-                    catButtons.setMessage(message.getIdLong());
-                    context.getData().addButtonGroup(context.getChannel(), message, catButtons);
-                });
+                if (message.getEmbeds().size() > 0) {
+                    EmbedBuilder embedBuilder = MessagingObjects.getClearThreadLocalEmbedBuilder();
+                    embedBuilder.setImage(getCatUrl());
+                    message.editMessage(embedBuilder.build()).queue();
+                } else {
+                    context.getUIMessaging().replyImage(getCatUrl()).thenAccept(catMessage -> {
+                        catButtons.addButtonsToMessage(catMessage);
+                        catButtons.setMessage(catMessage.getIdLong());
+                        context.getData().addButtonGroup(context.getChannel(), catMessage, catButtons);
+                    });
+                    message.delete().queue();
+                }
             } catch (IOException e) {
-                context.getTypedMessaging().replyDanger("Error loading cat picture " + UnicodeConstants.FROWNING);
+                message.editMessage("Error loading cat picture " + UnicodeConstants.FROWNING).queue();
```

Comment:
Why? This won't be a danger message?

Reply:
This is in the edit on the buttons

Reply:
this is in the refresh button, not the sending

Reply:
Should it not still return a danger embed?

Reply:
That would get complicated considering embeds are optional, but we could do. I'll think about this one a bit.

Reply:
Ngl I think we need to tidy up messaging a bit and get it to use a message builder so we use it for sending & editing

Reply:
how about we do that in a future pr, and w can get this merged for now